src/redux/tasks/actions/create.ts

```diff
@@ -0,0 +1,55 @@
+import { Dispatch } from '../../RootReducer'
+import { API } from '../../../services/http'
+
+import { ITask } from '../types'
+
+export const CREATE_TASK = 'CREATE_TASK'
+export const CREATE_TASK_SUCCESS = 'CREATE_TASK_SUCCESS'
+export const CREATE_TASK_ERROR = 'CREATE_TASK_ERROR'
+
+export type Actions = {
+    CREATE_TASK: {   
+        type: typeof CREATE_TASK,
+    },
+    CREATE_TASK_SUCCESS: {   
+        type: typeof CREATE_TASK_SUCCESS,
+        task: ITask,
+    },
+    CREATE_TASK_ERROR: {     
+        type: typeof CREATE_TASK_ERROR,
+        error: string
+    }
+}
+
+export const actionCreators = {
+    // --------------------------------------- //
+    //                    SYNC                 //
+    // --------------------------------------- //
+    createTaskRequest: ():  
+    Actions[typeof CREATE_TASK] => ({
+        type: CREATE_TASK,
+    }),
+    createTaskSuccess: (task: ITask):    
+    Actions[typeof CREATE_TASK_SUCCESS] => ({
+        type: CREATE_TASK_SUCCESS,
+        task,
+    }),
+    createTaskError: (error: string):    
+    Actions[typeof CREATE_TASK_ERROR] => ({
+        type: CREATE_TASK_ERROR,
+        error,
+    }),
+
+    // --------------------------------------- //
+    //                   ASYNC                 //
+    // --------------------------------------- //
+    createBackendTasks: (task: ITask) => {    
+        return (dispatch: Dispatch) => {
+            dispatch(actionCreators.createTaskRequest())
+            return API.post(`/checklists/${task.tasksList.id}/checkItems`, task).then(
```

Comment:
Really seems weird here, that task has the taskList in it, that has the tasks, etc...